Scripts/AwsEC2GetPublicSGRules/AwsEC2GetPublicSGRules.py

```diff
@@ -0,0 +1,138 @@
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+import json
+import copy
+
+
+def get_dict_value(data, key):
+    """ Returns dict value for a given key (case insensitive) """
+    for key_name in data.keys():
+        if key_name.lower() == key.lower():
+            return data[key_name]
+
+    return None
+
+
+def raise_error(error):
+    return {
+        'Type': entryTypes['error'],
+        'ContentsFormat': formats['text'],
+        'Contents': str(error)
+    }
+
+
+def get_ec2_sg_public_rules(group_id, ip_permissions, checked_protocol=None, checked_from_port=None,
+                            checked_to_port=None, region=None, include_ipv6='no'):
+    """
+        Get the list of public
+        which can be passed on to the following command:
+        aws-ec2-revoke-security-group-ingress-rule
+    """
+    public_rules = []
+    for rule in ip_permissions:
+        # Check protocol
+        protocol = get_dict_value(rule, 'IpProtocol')
+        if protocol != '-1':
+            if checked_protocol.lower() != protocol.lower():
```

Comment:
`checked_protocol` might be None and wont have `.lower()` attribute. same goes for `protocol` (if it's not in the rule it will be None).

Reply:
In theory, these could be none. Currently the assumption was:
- For `protocol`, since this script is intended to work with AWS IpPermissions, the rule will always have `IpProtocol`. 
- `checked_protocol` will always have value (it's set as required in the yml). If user try to run the script without specifying the value, the script won't even get executed due to the missing args, right?